Provisioning/Vault/vault-secrets-operator.tf

```diff
@@ -0,0 +1,70 @@
+resource "vault_auth_backend" "kubernetes" {
+  type = "kubernetes"
+}
+
+# Use local service account token and CA cert
+# https://developer.hashicorp.com/vault/docs/auth/kubernetes
+resource "vault_kubernetes_auth_backend_config" "kubernetes" {
+  backend            = vault_auth_backend.kubernetes.path
+  kubernetes_host    = "https://kubernetes.default.svc"
+  issuer             = "https://kubernetes.default.svc.cluster.local"
```

Comment:
- [ ] Todo: disable_iss_validation=true is the new recommended value for all versions of Vault (https://developer.hashicorp.com/vault/docs/auth/kubernetes#discovering-the-service-account-issuer)